src/PSDParametrizations/PSDParametrizations.jl

```diff
@@ -0,0 +1,29 @@
+
+abstract type PSDTrait end
+struct IsPSDParametrization <: PSDTrait end
+struct IsNotPSDParametrization <: PSDTrait end
+
+ispsdparametrization(::Any) = IsNotPSDParametrization()
+
+"""
+    psdparamtrization(::Type{T}, P)
+
+Wraps P in a psd paramtrization of eltype T.
+If P is already a type of psd paramtrization, then just the eltype is converted.
+"""
+function psdparamtrization(::Type{T}, P) where {T} end
```

Comment:
I think I'm quite confused to what this function does. Why is the eltype so important? Why do I as a user have to implement this one here and not `psdparameterization(P)`?

Reply:
Also there is a typo here, it shouldn't be `psdparamtrization`. Probably best to do a full search/replace for this

Reply:
Since any type representing a PSD could subtype ```Number```, ```AbstractMatrix```, ```Factorization``` or maybe none of the former, eltype can not be converted by ```convert(CommonSuperType{T}, psd_instance)```.  This would be a way to allow for eltype conversion uniformly, which seems useful to me.

However, currently it is also used to convert ```Number``` and ```AbstractMatrix``` into a self-adjoint type, i.e. taking the real part or wrapping in ```Symmetric``` or ```Hermitian``` depending on the eltype. Maybe this should be a separate type of function (at the cost of adding more (logic to) constructors)?

Reply:
## Changes from previous design in this PR

* ```IsPSD / IsNotPSD``` rather than ```IsPSDParametrization / IsNotPSDParametrization``` 
* ```psdcheck``` rather than  ```ispsdparametrization``` 
* ```convert_psd_eltype``` rather than ```psdparametrization``` 

## Untangling psdness and skedasticity 

New types for annotating skedasticity: 

```julia 
abstract type Skedasticity end
struct Homoskedastic <: Skedasticity end
struct Heteroskedastic <: Skedasticity end

_skedasticity(::IsPSD) = Homoskedastic()
_skedasticity(::IsNotPSD) = Heteroskedastic()

skedasticity(Σ) = _skedasticity(psdcheck(Σ))
```

## Tests 

...